### PR TITLE
 notebook for hierarchical Bayesian neural networks

### PR DESCRIPTION
**Pull request by murphyk**
Add notebook for hierarchical Bayesian neural networks for solving multiple related 2d binary classification problems.
This is a blackjax port of [this pymc3 notebook](https://github.com/twiecki/WhileMyMCMCGentlySamples/blob/master/content/downloads/notebooks/bayesian_neural_network_hierarchical.ipynb) by Thomas Wiecki. The porting was done by @karalleyna

### CODE REVIEW

**Comment by rlouf** [comment]
Thank you and @karalleyna for this contribution ! I am happy to see that `blackjax` work well with `distrax`.

1. I ran this on my machine and it seems slower than the numpyro port, was it the same for you?
2. Code looks great, I will need to add some additional text / explanations before merging if that's ok with you.

**Comment by murphyk** [comment]
I just updated the notebook with slightly more explanations :)

**Comment by rlouf** [comment]
Looks great ! Could you rename the file `HierarchicalBNN.ipynb` and squash your commits so I can merge?

File: notebooks/bnn_hierarchical_blackjax.ipynb [hunk 1]
```diff
@@ -0,0 +1,965 @@
...
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "0qS2Qh3_E6sv"
+      },
+      "source": [
+        "# Hierarchical Bayesian neural networks\n",
+        "\n",
+        "Code is based on  [This blog post](https://twiecki.io/blog/2018/08/13/hierarchical_bayesian_neural_network/) by Thomas Wiecki.\n",
```
Comment (twiecki) [inline notebooks/bnn_hierarchical_blackjax.ipynb:38]:
```suggestion
        "Code is based on  [This blog post](https://twiecki.io/blog/2018/08/13/hierarchical_bayesian_neural_network/) by [Thomas Wiecki](https://twitter.com/twiecki).\n",
```

**Comment by twiecki** [comment]
Very neat. @rlouf want to add ReviewNB?

**Comment by rlouf** [comment]
> Very neat. @rlouf want to add ReviewNB?

I did not expect so many example PRs, but it turns out this would be useful, yes.

**Comment by rlouf** [comment]
I am going to take care of the renaming. Merging. Thank you so much for contributing this example !

**Pull request merged**